Ingress/controllers/nginx-third-party/ssl/main.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ssl
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/golang/glog"
+)
+
+// Certificate contains the cert, key and the list of valid hostnames
+type Certificate struct {
+	Cert  string
+	Key   string
+	Cname []string
+	Valid bool
+}
+
+// CreateSSLCerts reads the content of the /etc/nginx-ssl directory and
+// verifies the cert and key and extracts the common names for this pair
+func CreateSSLCerts(baseDir string) []Certificate {
+	sslCerts := []Certificate{}
+
+	glog.Infof("inspecting directory %v for SSL certificates\n", baseDir)
+	files, _ := ioutil.ReadDir(baseDir)
+	for _, file := range files {
+		if !file.IsDir() {
+			continue
+		}
+
+		// the name of the secret could be different than the certificate file
+		cert := firstFile(fmt.Sprintf("%v/%v", baseDir, file.Name()), ".crt")
+		key := strings.Replace(cert, ".crt", ".key", 1)
```

Comment:
Can you perform a stronger check for a missing .key? and log it and document that the key and crt need to match, and we take the first crt etc.